### PR TITLE
Fix an issue in `haveEffectorsTrajectories`.

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
## Description

This PR robustify the test `contact_sequence.haveEffectorsTrajectories`.
In practice it works well except for:
- null transform.
- identity transform.

This is a well known issue from Eigen `isApprox` function. Hence the trick is to convert the two compared transforms into homogeneous matrices, subtract them and compare them to epsilon.

In essence I replace:
```
pinocchio SE3 a, b;
a.isApprox(b);
```
with:
```
pinocchio SE3 a, b;
(a.toHomogeneousMatrix() - b.toHomogeneousMatrix()).isMuchSmallerThan(1.0, prec)
```
Which is more verbose but more robust.

### CODE REVIEW

**Review by nim65s — APPROVED** [review]

**Comment by MaximilienNaveau** [comment]
@Guilhem there is no CI here?

**Comment by nim65s** [comment]
https://rainboard.laas.fr/ is down, this was escalated

**Comment by MaximilienNaveau** [comment]
So we should wait until the CI is back to merge I guess?

**Pull request merged**
